File: NxNGOL/src/mainer.java
import java.io.IOException;
import java.util.Random;

// main method in which we make use of the cell grid
public class mainer {
	
	///// WE DEFINE OUR NON-CONSTANT PARAMETERS /////
	public static final int WIDTH = 10;
	public static final int HEIGHT = 10;
	
	///// ENTRY POINT /////	
	public static void main(String[] args){
		
		// we define our non-constant parameters
		final int generations = 30;
		Random randomizer = new Random();
		
		// we define the seed of our game
		boolean[][] seed = new boolean[HEIGHT][WIDTH];
		for (int i = 0; i < HEIGHT; ++i){
			for(int j = 0; j < WIDTH; ++j){
				seed[i][j] = randomizer.nextBoolean();
			}
		}
		
		try{
			// we define our grid 
			grid map = new grid(seed, HEIGHT, WIDTH);
			
			// we print out the seed generation
			System.out.println("(Seed) Generation #0:");
			mapPrint(map);
					
			// we loop through our grid, each time updating 
			for (int i = 0; i < generations; ++i){
				System.out.println("Generation #" + (i+1) + ":");
				map.updateGrid();
				mapPrint(map);
			} // end of for loop

		}// end of try statement
		
		catch(IOException e){
			System.out.println("IOException when creating the grid!");
			e.printStackTrace();
		}// end of IO catch statement
		catch(NullPointerException e){
			System.out.println("NP exception when creating the grid!");
			e.printStackTrace();
		}// end of NP catch statement
		
	}// end of main method
	
	// method which prints out the current map
	private static void mapPrint(grid map){
		map.printGrid();
		System.out.println("*************");
		System.out.println("*************");
	}// end of mapPrint() method
	
}// end of main class

File: NxNGOL/src/mainer.py
#we import the necessary modules for the project
import grid as grid
import numpy as np
import random

#main method
def main():
    
    #we define our non-constant parameters
    generations = 20
    width = 5
    height = 5

    #we randomize the seed of the game
    seed = np.empty([width, height], dtype=bool)
    for i in range(width):
        for j in range(height):
            seed[i][j] = random.choice([True, False])

    #we define our grid
    gridMap = grid.grid(seed, height, width)

    #we print out the seed generation
    print("(Seed) Generation #0:")
    gridMap.printGrid()
    print("*************")
    print("*************")

    #we loop through our grid, each time updating
    for i in range(generations):
        print("Generation #", (i+1), ":")
        gridMap.updateGrid()
        gridMap.printGrid()
        print("*************")
        print("*************")
        

if __name__ == "__main__":
   main()


        
            
